crates/polars-python/src/series/scatter.rs

```diff
@@ -139,3 +140,74 @@ fn scatter_impl(
 
     s.and_then(|s| s.cast(&logical_dtype))
 }
+
+#[pymethods]
+impl PySeries {
+    /// Given a `PySeries` of length 0, find the index of the first value within
+    /// self.
+    fn index_of(&self, value: PySeries) -> PyResult<Option<usize>> {
+        // TODO assert length of value is 1?
+        index_of(&self.series, &value.series).map_err(|e| PyErr::from(PyPolarsErr::from(e)))
+    }
+}
+
+/// Try casting the value to the correct type, then call index_of().
+macro_rules! try_index_of {
+    ($self:expr, $value:expr) => {{
+        let cast_value = $value.map(|v| AnyValue::from(v).strict_cast($self.dtype()));
+        if cast_value == Some(None) {
+            // We can can't cast the searched-for value to a valid data point
+            // within the dtype of the Series we're searching, which means we
+            // will never find that value.
+            None
+        } else {
+            let cast_value = cast_value.flatten();
+            $self.index_of(cast_value.map(|v| v.extract().unwrap()))
+        }
+    }};
+}
+
+fn index_of(series: &Series, value_series: &Series) -> PolarsResult<Option<usize>> {
+    let value_series = if value_series.dtype().is_null() {
+        // Should be able to cast null dtype to anything, so cast it to dtype of
+        // Series we're searching.
+        &value_series.cast(series.dtype())?
+    } else {
+        value_series
+    };
+    let value_dtype = value_series.dtype();
+
+    if value_dtype.is_signed_integer() {
```

Comment:
(This should probably be part of the match below, in this state simply because code isn't cleaned up yet.)